src/engraving/dom/edit.cpp

```diff
@@ -1386,124 +1385,143 @@ void Score::cmdAddTimeSig(Measure* fm, staff_idx_t staffIdx, TimeSig* ts, bool l
+                bool changeActual = measure->ticks() == measure->timesig();
+                measure->undoChangeProperty(Pid::TIMESIG_NOMINAL, targetTimeSigFraction);
                 if (changeActual) {
-                    m->undoChangeProperty(Pid::TIMESIG_ACTUAL, ns);
+                    measure->undoChangeProperty(Pid::TIMESIG_ACTUAL, targetTimeSigFraction);
                 }
             }
-            std::pair<staff_idx_t, staff_idx_t> staffIdxRange = getStaffIdxRange(score);
+            const std::pair<staff_idx_t, staff_idx_t> staffIdxRange = getStaffIdxRange(score);
             for (staff_idx_t si = staffIdxRange.first; si < staffIdxRange.second; ++si) {
-                TimeSig* nsig = toTimeSig(seg->element(si * VOICES));
-                if (!nsig) {
+                TimeSig* timeSigOfCurrentStaffIndex = toTimeSig(targetSegment->element(si * VOICES));
+                if (!timeSigOfCurrentStaffIndex) {
                     continue;
                 }
-                nsig->undoChangeProperty(Pid::SHOW_COURTESY, ts->showCourtesySig());
-                nsig->undoChangeProperty(Pid::TIMESIG, ts->sig());
-                nsig->undoChangeProperty(Pid::TIMESIG_TYPE, int(ts->timeSigType()));
-                nsig->undoChangeProperty(Pid::NUMERATOR_STRING, ts->numeratorString());
-                nsig->undoChangeProperty(Pid::DENOMINATOR_STRING, ts->denominatorString());
-                nsig->undoChangeProperty(Pid::TIMESIG_STRETCH, ts->stretch());
-                nsig->undoChangeProperty(Pid::GROUP_NODES, ts->groups().nodes());
-                nsig->setSelected(false);
-                nsig->setDropTarget(false);
+                timeSigOfCurrentStaffIndex->undoChangeProperty(Pid::SHOW_COURTESY, newTimeSignature->showCourtesySig());
+                timeSigOfCurrentStaffIndex->undoChangeProperty(Pid::TIMESIG, newTimeSignature->sig());
+                timeSigOfCurrentStaffIndex->undoChangeProperty(Pid::TIMESIG_TYPE, int(newTimeSignature->timeSigType()));
+                timeSigOfCurrentStaffIndex->undoChangeProperty(Pid::NUMERATOR_STRING, newTimeSignature->numeratorString());
+                timeSigOfCurrentStaffIndex->undoChangeProperty(Pid::DENOMINATOR_STRING, newTimeSignature->denominatorString());
+                timeSigOfCurrentStaffIndex->undoChangeProperty(Pid::TIMESIG_STRETCH, newTimeSignature->stretch());
+                timeSigOfCurrentStaffIndex->undoChangeProperty(Pid::GROUP_NODES, newTimeSignature->groups().nodes());
+                timeSigOfCurrentStaffIndex->setSelected(false);
+                timeSigOfCurrentStaffIndex->setDropTarget(false);
+            }
+            if (local) {
+                select(originalTimeSig, SelectType::SINGLE, staffIdx);
+            } else {
+                selectElementsWithSameTypeOnSegment(ElementType::TIMESIG, targetSegment);
             }
```

Comment:
Perhaps the reason that this does not seem to work is that it is only inside the `if (measureDurationUnchanged)` case, and not in the `else` case.